testkit/src/main/java/org/apache/calcite/test/CalciteAssert.java

```diff
@@ -569,6 +569,13 @@ static void assertQuery(
       try {
         if (updateChecker == null) {
           resultSet = statement.executeQuery(sql);
+          if (resultChecker == null && exceptionChecker != null) {
```

Comment:
This is a separate bug, which was uncovered by the `@NonDeterministic` annotation on toInt. The exception was happening at compilation time, instead of runtime. This change is necessary to uncover exceptions that happen at runtime.